src/GafferImage/ImageWriter.cpp

```diff
@@ -710,28 +1065,38 @@ void ImageWriter::execute() const
 		}
 	}
 
-	const int nChannels = maskChannels.size();
+	Format imageFormat = inPlug()->formatPlug()->getValue();
 
-	// Get the image channel data.
-	IECore::ImagePrimitivePtr imagePtr( inPlug()->image() );
+	/// \todo Remove when Issue #887 is done
+	if( imageFormat.getDisplayWindow().isEmpty() )
+	{
+		imageFormat = FormatPlug::getDefaultFormat( Context::current() );
+	}
 
-	// Get the image's display window.
-	const Imath::Box2i displayWindow( imagePtr->getDisplayWindow() );
-	const Imath::Box2i dataWindow( imagePtr->getDataWindow() );
+	Imath::Box2i dataWindow = inPlug()->dataWindowPlug()->getValue();
+	Imath::Box2i exrDataWindow( Imath::V2i( 0 ) );
 
-	ImageSpec spec = createImageSpec( this, out.get(), dataWindow, displayWindow );
+	if( !empty( dataWindow ) )
+	{
+		exrDataWindow = imageFormat.toEXRSpace( dataWindow );
```

Comment:
Oh wait, this isn't a pass though, is it. Its taking an un-initialized window and replacing it with a (0,0,->(0,0) window.